controllers/controller/devworkspacerouting/solvers/basic_solver.go

```diff
@@ -43,12 +43,38 @@ var nginxIngressAnnotations = func(endpointName string, endpointAnnotations map[
 	return annotations
 }
 
+func serviceAnnotations(sourceAnnotations map[string]string, isDiscoverable bool, serviceRoutingConfig controllerv1alpha1.Service) map[string]string {
+	annotations := make(map[string]string)
+	if sourceAnnotations != nil && len(sourceAnnotations) > 0 {
+		for k, v := range sourceAnnotations {
+			annotations[k] = v
+		}
+	}
+	if isDiscoverable {
+		annotations[constants.DevWorkspaceDiscoverableServiceAnnotation] = "true"
+	}
+	if serviceRoutingConfig.Annotations != nil && len(serviceRoutingConfig.Annotations) > 0 {
+		for k, v := range serviceRoutingConfig.Annotations {
+			annotations[k] = v
+		}
+	}
+	return annotations
+}
+
 // Basic solver exposes endpoints without any authentication
 // According to the current cluster there is different behavior:
 // Kubernetes: use Ingresses without TLS
 // OpenShift: use Routes with TLS enabled
 type BasicSolver struct{}
 
+var routingSuffixSupplier = func() string {
+	return config.GetGlobalConfig().Routing.ClusterHostSuffix
+}
+
+var isOpenShift = func() bool {
+	return infrastructure.IsOpenShift()
```

Comment:
Since `infrastructure.IsOpenShift()` is commonly used in the codebase, I don't think this fcuntion is necessary

Reply:
Actually, I had exposed this for testing. I wanted to mock the behavior of `infrastructure.IsOpenShift()` method. 

Let me think on how I can do it without adding this function.